module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,50 @@
+#include "compression.hpp"
+
+compressed_bitmap compressGrayscale(const bitmap& image) {
+    compressed_bitmap compressedImage{};
+    compressedImage.reserve(height * width);
+
+    uint8_t currentGrayscale;
+    uint8_t counter;
+    uint8_t grayscale;
+
+    for (size_t i = 0; i < height; ++i) {
+        currentGrayscale = image[i][0];
+        counter = 1;
+        for (size_t j = 1; j < width; ++j) {
+            grayscale = image[i][j];
+            if (grayscale == currentGrayscale) {
+                counter++;
+            } else {
+                compressedImage.push_back({currentGrayscale, counter});
+                currentGrayscale = grayscale;
+                counter = 1;
+            }
+        }
+        compressedImage.push_back({currentGrayscale, counter});
```

Comment:
Try to reconfigure your loop, to call push_back only once.

Reply:
Done, however is still ugly, can you recommend something? I have been thinking for a while, made few solutions but every has the same problem :(